### PR TITLE
[DependencyInjection] Inject defaults and instanceof conditionals in anonymous services

### PR DESCRIPTION
**Pull request by GuilhemN**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/21970#issuecomment-286351183
| License       | MIT
| Doc PR        | 

This PR injects defaults and instanceof conditionals in anonymous services to be consistent for all services.

```yml
services:
    _defaults:
        autowire: true

    Bar: [ !service { class: Quz } ]
```
In this example, `Bar` and the anonymous service will be autowired.

One special case: instanceof conditionals aren't injected in anonymous services used in instanceof conditionals.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 1]
```diff
@@ -280,11 +281,7 @@ private function parseDefinition(\DOMElement $service, $file, array $defaults =
             if ($function = $configurator->getAttribute('function')) {
                 $definition->setConfigurator($function);
             } else {
-                $configuratorService = $this->getChildren($configurator, 'service');
-
-                if (isset($configuratorService[0])) {
-                    $class = $this->parseDefinition($configuratorService[0], $file);
```
Comment (GuilhemN) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: I fixed another bug at the same time, the fact that the factories/configurators were inline meant for prototypes/instanceof conditionals that they were not always the same instance which wasn't consistent with the rest of the anonymous services.
Note that it wasn't an issue before as a service was never cloned.

**Comment by nicolas-grekas** [comment]
rebase needed

**Comment by GuilhemN** [comment]
@nicolas-grekas done

**Comment by nicolas-grekas** [comment]
Thinking again about that, what about going the other way around: have defaults and instanceof apply only to root level definitions, and not to nested (inline) ones?
That'd be my preference.
If you agree, let's close this PR and work on another one.

**Comment by GuilhemN** [comment]
> Thinking again about that, what about going the other way around: have defaults and instanceof apply only to root level definitions, and not to nested (inline) ones?

That's the current behavior.

> That'd be my preference.
> If you agree, let's close this PR and work on another one.

I agree, thinking about it again I don't think this change is worth it. Let's close it and open a different PR for https://github.com/symfony/symfony/pull/21999#discussion_r106019873.

**Pull request closed without merging**